spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -267,6 +372,20 @@ data class SpdxLicenseIdExpression(
         } ?: throw SpdxException("'$this' is not a valid SPDX license id.")
     }
 
+    override fun equals(other: Any?) =
+        when (other) {
+            is SpdxLicenseIdExpression -> id == other.id && orLaterVersion == other.orLaterVersion
+            is SpdxExpression -> {
```

Comment:
Nit: For consistency with the other `when`s where either case is a block we'd also need a blank line here.